### PR TITLE
git-instafix: init at 0.2.1

### PR DESCRIPTION
**Pull request by mightyiam**
## Description of changes

This app seems nice. Here's a package for everyone!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gi/git-fixup/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}:
+let
+  inherit
+    (lib)
+    licenses
+    maintainers
+    ;
+
+  version = "0.1.8";
+  pname = "git-fixup";
```
Comment (jfly) [inline pkgs/by-name/gi/git-fixup/package.nix:14]: Why the separate binding for `pname`? It seems to only be used on 1 place.

**Review by jfly — APPROVED** [review]
:shipit:

**Comment by rjpcasalino** [comment]
Result of `nixpkgs-review pr 275402` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>git-fixup</li>
  </ul>
</details>

```
[nix-shell:~/.cache/nixpkgs-review/pr-275402/results/git-fixup/bin]$ ./git-fixup --version
git-fixup 0.1.8
```

File: pkgs/by-name/gi/git-fixup/package.nix (hunk 1 above)
Comment (rjpcasalino) [reply pkgs/by-name/gi/git-fixup/package.nix:14]: I think that's just the convention? :shrug:

**Review by rjpcasalino — APPROVED** [review]

File: pkgs/by-name/gi/git-fixup/package.nix (hunk 1 above)
Comment (mightyiam) [reply pkgs/by-name/gi/git-fixup/package.nix:14]: Just an oversight, I suppose.

**Comment by kuruczgy** [comment]
Could you talk a bit about this particular git-fixup project? It is made mostly by a single author, with the last commit from more than a year ago. Contrast this with https://github.com/keis/git-fixup, which seems to be receiving semi-regular updates, with multiple contributors.

I would propose either
- switching to https://github.com/keis/git-fixup unless you depend on some specific feature of this rust version,
- or renaming this to something else, so that if someone in the future wants to package https://github.com/keis/git-fixup, it can get the `git-fixup` name.

**Comment by mightyiam** [comment]
[Author considering renaming their project](https://github.com/quodlibetor/git-fixup/pull/11). I'd give them some more time.

**Comment by kuruczgy** [comment]
It seems that the author renamed the project, did a new release with the new name, and even provided a description of what differentiates it from the shell-based version I linked, so from my side this can go ahead :+1: (As long as you adjust this PR with the new name and the new version of course.)

**Comment by mightyiam** [comment]
Made all the updates.

File: pkgs/by-name/gi/git-instafix/package.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  buildInputs = [ libgit2 ];
+
+  nativeCheckInputs = [ git ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postFixup = ''
+    wrapProgram $out/bin/git-instafix --prefix PATH : ${lib.makeBinPath [ git ]}
+  '';
```
Comment (kuruczgy) [inline pkgs/by-name/gi/git-instafix/package.nix:47]: Ideally we should ask the author to make the git binary configurable. But for the first version this should be fine.

File: pkgs/by-name/gi/git-instafix/package.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{ git
+, lib
+, libgit2
+, makeWrapper
+, rustPlatform
+, fetchFromGitHub
+,
```
Comment (kuruczgy) [inline pkgs/by-name/gi/git-instafix/package.nix:7]: Nitpick: unnecessary trailing comma.

File: pkgs/by-name/gi/git-instafix/package.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  cargoHash = "sha256-3l400AUoDmYRQQWU4VtUS/PJHoJ/Y0RGSy4N3nnH2ok=";
+
+  meta = {
+    description = "Quickly fix up an old commit using your currently-staged changes.";
+    homepage = "https://github.com/quodlibetor/git-instafix";
+    license = with licenses; [ mit asl20 ];
+    maintainers = with maintainers; [ mightyiam ];
+    changelog = "https://github.com/quodlibetor/git-instafix/releases/tag/v${version}";
+  };
```
Comment (kuruczgy) [inline pkgs/by-name/gi/git-instafix/package.nix:37]: `meta` should be at the very bottom, at least that's what I see in most packages.

**Review by kuruczgy — APPROVED** [review]
LGTM, just minor formatting nitpicks.

**Comment by mightyiam** [comment]
Applied feedback.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 275402` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>git-instafix</li>
  </ul>
</details>


Ofborg darwin seem to be failing due to a linker error, consider marking it `meta.broken = stdenv.isDarwin`

**Review by pbsds — COMMENTED** [review]
still broken on darwin

**Comment by mightyiam** [comment]
Thank you!

**Pull request merged**